### PR TITLE
Match audio_seqplayer.c

### PR DESCRIPTION
**Pull request by Synray**

### CODE REVIEW

**Comment by ethteck** [comment]
Awesome

**Review by ethteck — APPROVED** [review]

File: src/code/audio_seqplayer.c [hunk 1]
```diff
@@ -1455,25 +1452,18 @@ void Audio_SequenceChannelProcessScript(SequenceChannel* channel) {
...
     M64ScriptState* seqScript = &seqPlayer->scriptState;
     s16 tempS;
     u16 temp;
     s32 i;
     s32 value;
     u8* data;
-    s32 pad1;
-    s32 pad2;
+    u8* data2; // force a1
+    u8* data3; // force v0
```
Comment (Dragorn421) [inline src/code/audio_seqplayer.c:1466]: These comments can be removed?

File: src/code/audio_seqplayer.c [hunk 2]
```diff
@@ -1322,20 +1320,19 @@ void Audio_SequenceChannelProcessScript(SequenceChannel* channel) {
                         channel->unk_22 = ((u16*)(channel->dynTable))[scriptState->value];
                         break;
                     case 0xB6:
-                        data = (*channel->dynTable)[0];
                         scriptState->value = (*channel->dynTable)[0][scriptState->value];
                         break;
                     case 0xB7:
-                        channel->unk_22 = (parameters[0] == 0) ? gAudioContext.audioRandom // odd load here
+                        channel->unk_22 = (parameters[0] == 0) ? gAudioContext.audioRandom & 0xffff
```
Comment (Dragorn421) [inline src/code/audio_seqplayer.c:1326]:
```suggestion
                        channel->unk_22 = (parameters[0] == 0) ? gAudioContext.audioRandom & 0xFFFF
```
and other 0xffff too

**Review by Dragorn421 — COMMENTED** [review]
Nice

I'm interested in what ido decomp brought, regalloc logic?

File: src/code/audio_seqplayer.c [hunk 3]
```diff
@@ -1195,7 +1193,7 @@ void Audio_SequenceChannelProcessScript(SequenceChannel* channel) {
                     case 0xCB:
                         offset = (u16)parameters[0];
 
-                        scriptState->value = *(offset + scriptState->value + player->seqData);
+                        scriptState->value = *(player->seqData + (u32)(offset + scriptState->value));
```
Comment (fig02) [inline src/code/audio_seqplayer.c:1196]: not related to your set of changes, and we dont need to do this now if you dont want to, but at some point we really need to changes all instances of `player->` here to `seqPlayer->` lmao, super confusing with the actual player

**Review by fig02 — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

File: src/code/audio_seqplayer.c [hunk 4]
```diff
@@ -1313,29 +1311,28 @@ void Audio_SequenceChannelProcessScript(SequenceChannel* channel) {
...
                     case 0xB5:
                         channel->unk_22 = ((u16*)(channel->dynTable))[scriptState->value];
                         break;
                     case 0xB6:
-                        data = (*channel->dynTable)[0];
                         scriptState->value = (*channel->dynTable)[0][scriptState->value];
                         break;
                     case 0xB7:
-                        channel->unk_22 = (parameters[0] == 0) ? gAudioContext.audioRandom // odd load here
+                        channel->unk_22 = (parameters[0] == 0) ? gAudioContext.audioRandom & 0XFFFF
```
Comment (Dragorn421) [inline src/code/audio_seqplayer.c:1326]:
```suggestion
                        channel->unk_22 = (parameters[0] == 0) ? gAudioContext.audioRandom & 0xFFFF
```
😅

Comment (AngheloAlf) [reply src/code/audio_seqplayer.c:1326]: `X`

Comment (fig02) [reply src/code/audio_seqplayer.c:1326]: i will just use githubs commit feature to make this easier for everyone lol

**Comment by fig02** [comment]
there are more in the file that need to change, so Ill leave that to you synray

**Review by Dragorn421 — APPROVED** [review]

**Comment by Synray** [comment]
replying to dragorn, based on the registers that were different, I could sort of narrow down what I needed to try. the first function had regalloc problems caused by uopt (temp vars), and the second one mostly had problems caused by ugen (masks and type conversions etc.)

basically regalloc starts at `$v0` and goes up, so since `data` was given `$v1`, something was interfering with `$v0`, like `data` having lower priority, or another var already using `$v0`

I stepped through in gdb and tried to figure out what specifically was causing the diffs, but my print functions aren't quite good enough for that yet.

**Review by Roman971 — APPROVED** [review]

**Pull request merged**